include/deal.II/dofs/dof_tools.h

```diff
@@ -1697,6 +1697,82 @@ namespace DoFTools
   extract_locally_relevant_dofs(const DoFHandlerType &dof_handler,
                                 IndexSet &            dof_set);
 
+
+  /**
+   * Extract the set of local, active or relevant DoF indices for each
+   * components within the mask that are owned by the current  processor. For
+   * components disabled by the mask, an empty IndexSet is returned.
+   * For regular DoFHandler objects, this set is the complete set with all DoF
+   * indices. In either case, it equals what DoFHandler::locally_owned_dofs()
+   * returns. The ownership type is fixed by a small enum that controls which
+   * type of ownership is asked off.
+   */
+
+  enum class OwnershipType
+  {
+    owned,
+    active,
```

Comment:
We've generally tried to avoid introducing these one-time enums. I'd favor us using three different functions (like the ones we already have), or if you must an argument that is simply an integer or a string -- but three different functions would still be my preference.

Reply:
This would lead to three identical implementation, with the exception of a check on what cells to ask dofs to. They are one-time enums, if no-one else uses them... but there are many other functions that may actually be converted to a single one taking such an enum: it is not a function specific enum.

If you still prefer three different functions, we'll replicate them.